homeassistant/components/garmin_connect/__init__.py

```diff
@@ -0,0 +1,104 @@
+"""The Garmin Connect integration."""
+import asyncio
+from datetime import date, timedelta
+import logging
+
+from garminconnect import (
+    Garmin,
+    GarminConnectAuthenticationError,
+    GarminConnectConnectionError,
+    GarminConnectTooManyRequestsError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.util import Throttle
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+MIN_SCAN_INTERVAL = timedelta(minutes=10)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Garmin Connect component."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Garmin Connect from a config entry."""
+    username = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+
+    try:
+        garmin_client = Garmin(username, password)
```

Comment:
It's a better pattern to not have the class instantiation do side effects. Instead have the library user call `garmin_client.login()` to log in. Actions should be explicit.

Reply:
The communication flow with the portal is not standard, the Garmin init class does a get_full_name to get a displaystring (an id to be used to fetch data), so the login call is only called first time, and when session expires, the rest of the communication is done using a single get_stats().
Even when you provide wrong credentials, the portal also return status code 200, (not unauth) but the returned html doesn't contain the data needed.

For example the Toomanyrequests exception can occur at each call, so I implemented all exceptions on all calls.
I will try to rewrite the library so we call .login() instead if the Garmin class itself.
And use same or less api calls.

Reply:
I have now separated the Garmin.login method from the Class init, and fixed the exception catching. If anything else let me know.
Btw it would be nice to be able to also link integration/sensors to a Person instead of an Area using the GUI.

Reply:
But now the test_config_flows fail again, this test code has cost me double the time it took to wrote the whole integration includeing the python module. Very frustrating.